run/service-auth/receive_test.py

```diff
@@ -101,19 +124,75 @@ def services():
     )
 
 
-def test_auth(services):
-    url = services[0]
-    token = services[1]
+def test_authentication_on_cloud_run(service: tuple[str, str]) -> None:
+    endpoint_url = service[0]
```

Comment:
You're re-declaring this at the start of each of your tests. You could create a fixture that returns this endpoint for you. (Particularly because the `0` here could be a 'magic number', as Cloud Run services now have multiple URLs